src/EditorFeatures/Test/Extensions/SourceTextContainerExtensionsTests.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Microsoft.CodeAnalysis.Text;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.Extensions
+{
+    public class SourceTextContainerExtensionsTests
+    {
+        [Fact]
+        public void GetBufferTextFromNonTextContainerThrows()
+        {
+            var containerMock = new Mock<SourceTextContainer>();
+            Assert.Throws<ArgumentException>(containerMock.Object.GetTextBuffer);
+        }
+
+        [Fact]
+        public void GetBufferTextFromTextContainerReturnsTextBuffer()
```

Comment:
I had set out to test that the GetTextBuffer method works and returns the buffer that was supplied to the TextBufferContainer. I suppose that although at present the text buffer container does not mutate the buffer it may be a valid scenario for a future revision or even for a derived type. In which case, I will alter the test to assert only that GetTextBuffer does not throw.